src/objects/job.py

```diff
@@ -472,3 +479,99 @@ def _find_test_failures(self, junit_dir: str) -> list[Failure]:
             )
 
         return failures
+
+    def _get_timestamp(
+        self,
+        job_name: Optional[str],
+        build_id: Optional[str],
+    ) -> Optional[int]:
+        """
+        Fetches the UNIX timestamp of job
+        
+        Args:
+            job_name (Optional[str]): The name of the job to get timestamp for.
+            build_id (Optional[str]): The build ID of the job to get timestamp for.
+
+        Returns:
+            Optional[int]: The timestamp if found, otherwise None.
+        """
+
+        client = storage.Client()
+        bucket = client.bucket(self.gcs_bucket)
+        blob = bucket.blob(f"{job_name}/{build_id}/started.json")
+
+        try:
+            started_json = json.loads(blob.download_as_text())
+            return started_json.get("timestamp")
+        except Exception as e:
+            self.logger.error(f"Failed to fetch timestamp from started.json: {e}")
+            return None
+
+    def _get_all_build_ids(
+        self,
+        job_name: Optional[str],
+    ) -> list[str]:
+        """
+        Get the list of all build IDs from the job directory.
+        
+        Args:
+            job_name (Optional[str]): The name of the job to get timestamp for.
+    
+        Returns:
+            List[str]: A list of build IDs.
+        """
+        client = storage.Client()
+        bucket = client.bucket(self.gcs_bucket)
+        prefix = f"logs/{job_name}/"
+        blobs = bucket.list_blobs(prefix=prefix, delimiter='/')
+
+        build_ids = []
+        for page in blobs.pages:
+            build_ids.extend(page.prefixes)
+
+        # Get the build IDs from the prefixes
+        build_ids = [prefix.split('/')[-2] for prefix in build_ids if prefix.endswith('/')]
+        self.logger.info(f"Build IDs retrieved for {job_name}: {build_ids}")
+        return build_ids
+
+    def _check_is_retriggered(
+        self,
+        job_name: Optional[str],
+        build_id: Optional[str],
+    ) -> bool:
+        """Check if the current job build is retriggered within the same week as a previous build.
+
+        Args:
+            job_name (Optional[str]): The name of the job to get timestamp for.
+            build_id (Optional[str]): The build ID of the job to get timestamp for.
+
+        Returns:
+            bool: True if retriggered within the same week, False otherwise.
+        """
+        current_timestamp = self._get_timestamp(job_name, build_id)
+        if current_timestamp is None:
+            self.logger.error(f"Failed to get timestamp for job {job_name} with build ID {build_id}")
+            return False
+
+        current_datetime = datetime.fromtimestamp(current_timestamp, tz=timezone.utc)
+
+        # Calculate the start of the current week (Monday, 00:00 UTC)
+        week_start = current_datetime - timedelta(days=current_datetime.weekday())
+        week_start = week_start.replace(hour=0, minute=0, second=0, microsecond=0)
```

Comment:
I'm not sure if this is helpful, but you may want to verify that the way this resolves matches what your team considers the start of the week.

I'm just concerned that somehow this could catch the initial run from the week before or something. Not 100% sure, but it would be worth verifying, if you haven't already.

Reply:
Thanks, I tested with unit test mocking with some combinations, I have Monday 00:00 UTC as start of the week, but I can test with previous weeks entry as well

Reply:
@calebevans I have also updated the unit test mocking example, tested with previous weeks timestamp & works as expected